asn/asn.go

```diff
@@ -0,0 +1,60 @@
+package asn
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	asnmap "github.com/projectdiscovery/asnmap/libs"
+	"github.com/projectdiscovery/mapcidr"
+)
+
+type ASNClient struct {
+	client *asnmap.Client
+}
+
+func New() ASNClient {
+	return ASNClient{
+		client: asnmap.NewClient(),
+	}
+}
+
+// GetCIDRsForASNNum returns the slice of cidrs for given ASN number
+// accept the ASN number like 'AS15133' and returns the CIDRs for that ASN
+func (c *ASNClient) GetCIDRsForASNNum(value string) []*net.IPNet {
+	var cidrs []*net.IPNet
+	asn := asnmap.ASN(value[2:]) //drop the AS suffix
```

Comment:
- Let's check the length of `value` to avoid going out of slice bounds
- Comments are preferred above the line of code and with a space between markers and text (`// drop`)